app/notice_board/views.py

```diff
@@ -0,0 +1,47 @@
+from rest_framework.decorators import action
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+
+from notice_board.models import Meeting, MeetingParticipant
+from notice_board.serializers import MeetingSerializer
+
+
+class MeetingViewSet(ModelViewSet):
+
+    queryset = Meeting.objects.all()
+    serializer_class = MeetingSerializer
+
+    # pylint: disable=unused-argument
+    @action(detail=True, methods=["post"])
+    def add_participant(self, request: Request, pk=None):
+        meeting = self.get_object()
+        if meeting.participants_limit < 1:
+            meeting.participants.add(
+                request.user, through_defaults={"status": MeetingParticipant.CONFIRMED}
+            )
+            return Response({"status": "confirmed"})
+        MeetingParticipant.objects.create(
+            meeting=meeting,
+            user=request.user,
+            status=MeetingParticipant.WAITING,
+        )
+        status = MeetingParticipant.WAITING
+        participants = MeetingParticipant.objects.filter(
+            meeting=meeting,
+        ).order_by("created_at")
+        for i, participant in enumerate(participants):
+            if i >= meeting.participants_limit:
+                break
+            if participant.user == request.user:
+                participant.status = MeetingParticipant.CONFIRMED
+                status = participant.status
+                participant.save()
+        return Response({"status": status.lower()})
+
+    # pylint: disable=unused-argument
+    @action(detail=True, methods=["post"])
+    def remove_participant(self, request: Request, pk=None):
+        meeting = self.get_object()
+        meeting.participants.remove(request.user)
```

Comment:
Atfer you remove a participant you need to check i there is anyone that can pop in for his place.
You could reuse code from add participant action.